docs/src/whatsnew/latest.rst

```diff
@@ -91,7 +91,8 @@ This document explains the changes made to Iris for this release
 💣 Incompatible Changes
 =======================
 
-#. N/A
+#. `@bouweandela`_ updated cube comparison so cubes with data containing a
+   :obj:`numpy.nan` are now considered equal (:pull:`5713`)
```

Comment:
```suggestion
#. `@bouweandela`_ updated :class:`~iris.cube.Cube` comparison so equality is
   now possible between cubes with data containing a :obj:`numpy.nan`.
   e.g. ``Cube([np.nan, 1.0]) != Cube([np.nan, 2.0])``,
   ``Cube([np.nan, 1.0]) == Cube([np.nan, 1.0])``. (:pull:`5713`)
```

I'm worried about misinterpretation that ALL `Cube`s including `NaN` will be considered equal.